include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -115,38 +115,37 @@ namespace VectorTools
     // vectorfunction multiple times at the
     // same point on a cell.
     //
-    // note that we have to set up all of the
-    // following arrays for each of the
-    // elements in the FECollection (which
-    // means only once if this is for a regular
-    // DoFHandler)
-    std::vector<std::vector<Point<dim> > > unit_support_points (fe.size());
+    // First check that the desired components are interpolating.
     for (unsigned int fe_index=0; fe_index<fe.size(); ++fe_index)
       {
-        unit_support_points[fe_index] = fe[fe_index].get_unit_support_points();
-        Assert ((unit_support_points[fe_index].size() != 0)||(fe[fe_index].dofs_per_cell == 0),
-                ExcNonInterpolatingFE());
-      }
+        for (unsigned int component_index = 0; component_index < n_components; ++component_index)
+          {
+            if (component_mask[component_index] == true)
+              {
+                std::pair<unsigned int, unsigned int> base_index = fe[fe_index].component_to_base_index(component_index);
 
+                AssertThrow ((fe[fe_index].base_element(base_index.first).has_support_points()) ||
+                             (fe[fe_index].base_element(base_index.first).dofs_per_cell == 0),
+                             ExcNonInterpolatingFE());
+              }
+          }
+      }
 
     // Find the support points on a cell that
-    // are mentioned multiple times in
-    // unit_support_points.  Mark the first
-    // representative of each support point
-    // mentioned multiple times by appending
-    // its dof index to dofs_of_rep_points.
+    // are mentioned multiple times, and ony add
+    // each once.
     // Each multiple point gets to know the dof
     // index of its representative point by the
     // dof_to_rep_dof_table.
 
-    // the following vector collects all dofs i,
+    // the following vector collects all unit support points p[i],
     // 0<=i<fe.dofs_per_cell, for that
```

Comment:
that's an artifact of the person writing it having German as mother language. I think it should actually be "for which". Would you agree?